### PR TITLE
[Fix-11866] [API] When a user accesses an interface without project permission, the returned information is not formatted

### LINKED ISSUE #11866: [Bug] [API] When a user accesses an interface without project permission, the returned information is not formatted
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

When a user accesses an interface without project permission, the returned information is not formatted

Interface return information:
```
{
  "code": 30002,
  "msg": "当前用户[{0}]没有[{1}]项目的操作权限",
  "data": null,
  "failed": true,
  "success": false
}

```

### What you expected to happen

Interface return information:
```
{
  "code": 30002,
  "msg": "当前用户[zhangsan]没有[test]项目的操作权限",
  "data": null,
  "failed": true,
  "success": false
}

```

### How to reproduce

User access interface without project permission.

### Anything else

_No response_

### Version

3.0.0

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhuangchong**
## Purpose of the pull request

this pr close: #11866 

## Verify this pull request

This pull request is already covered by existing tests.

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java [hunk 1]
```diff
@@ -206,18 +203,17 @@ private void updateProcessDefiniteVersion(User loginUser, Map<String, Object> re
...
+
+        Map<String, Object> result = new HashMap<>();
         if (taskCode == 0) {
             logger.error("Delete task process relation error due to parameter taskCode is 0, projectCode:{}, processDefinitionCode:{}.",
                     projectCode, processDefinitionCode);
             putMsg(result, Status.DELETE_TASK_PROCESS_RELATION_ERROR);
             return result;
         }
+        // check user access for project
+        projectService.hasProjectAndPerm(loginUser, projectCode, null);
```
Comment (WangJPLeo) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java:306]: The current deletion operation and item data permission should be verified here and I'll improve this later.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/SchedulerServiceImpl.java [hunk 2]
```diff
@@ -143,10 +146,7 @@ public Map<String, Object> insertSchedule(User loginUser,
         Project project = projectMapper.queryByCode(projectCode);
 
         // check project auth
-        boolean hasProjectAndPerm = projectService.hasProjectAndPerm(loginUser, project, result, null);
-        if (!hasProjectAndPerm) {
-            return result;
-        }
+        projectService.hasProjectAndPerm(loginUser, project, null);
```
Comment (WangJPLeo) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/SchedulerServiceImpl.java:159]: Permission key should not be null, also fix them in next pr.

**Review by WangJPLeo — DISMISSED** [review]
LGTM

**Comment by EricGao888** [comment]
May I ask whether we are going to put it to 3.0.2? If not, I'm going to label it with 3.0.3.

**Comment by zhuangchong** [comment]
> May I ask whether we are going to put it to 3.0.2? If not, I'm going to label it with 3.0.3.

done.

**Comment by zhongjiajie** [comment]
I think we should better combine our project permission check into one, I used to do it at https://github.com/apache/dolphinscheduler/pull/12003, but at last I do not finish it yet

**Comment by zhongjiajie** [comment]
Maybe we should make it target to version 3.2.0, WDYT?

**Pull request closed without merging**